bundles/io/org.eclipse.smarthome.io.rest.core/src/main/java/org/eclipse/smarthome/io/rest/core/discovery/InboxResource.java

```diff
@@ -95,10 +95,12 @@ public Response approve(@PathParam("thingUID") @ApiParam(value = "thingUID", req
         if (results.isEmpty()) {
             return Response.status(Status.NOT_FOUND).build();
         }
+        Thing thing = inbox.approve(thingUIDObject,  label != null && !label.isEmpty() ? label : null);
         DiscoveryResult result = results.get(0);
```

Comment:
Do I understand the inbox.approve(..) code correctly, that we can drop the "result" and using thing.getThingTypeUID() in the calls below.

As long as inbox.approve already uses the InboxFilterCriteria, couldn't we move the "check if the thingUID" is part of the inbox in the approve implementation and throw an exception, catch them and return Status.NOT_FOUND?